message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -139,6 +139,17 @@
                 <param index="7">Empty</param>
             </entry>
 
+            <entry name="MAV_CMD_DO_ARM_CHECK" value="42429">
+                <description>Reply with whether or not you can arm and why</description>
```

Comment:
We've talked about a more robust method of reporting pre-arm status and this led to an attempted implementation in which we send a bitmask (see link below).  The implementation is ok but there was some discussion within the dev team that led to scope creep and the strive for perfection became the enemy of the good meaning it hasn't made it to master.
https://github.com/rmackay9/rmackay9-ardupilot/commits/aparm9